draft-ietf-rats-corim.md

```diff
@@ -2089,112 +2063,139 @@ Logically, new ECT entries are appended to the existing ACS.
 
-This specification defines the comparison algorithm for the codepoints described in sub-sections below.
-A CoRIM profile may define additional negative codepoints and their matching algorithms.
-Specifications that extend CoMID MUST also define comparison algorithms for their added codepoints.
-If a codepoint's comparison algorithm is not stated or does not default to the comparison algorithms described in this specification, then the Verifier MUST compare the binary equality of the CBOR encodings of the values.
+The algorithm used to compare a condition ECT against the ACS is stateless, it depends only on the condition ECT being compared and the contents of the ACS at the time of the comparison
+
+The Verifier uses a temporary "candidate entries" array as part of the comparision. Each element in the array is an ECT.
+
+After completing the steps described in this sub-section, the algorithm returns false if the candidate entries array is empty, true if it contains one or more elements.
 
 ### Environment Comparison {#sec-compare-env}
 
+The verifier SHALL iterate over all entries in the ACS and SHALL add each entry where the condition ECT stateful enviromnment `environment-map` is a subset of the ACS entry `environment-map` to the candidate entries array.
+
+The condition ECT `environment-map` is a subset of an ACS entry `environment-map` if each field (for example `class`, `instance` etc.) which is present in the condition ECT `environment-map` is also present in the ACS entry, and the CBOR encoded field values in the stateful environment and ACS entry are binary identical.
+If a field is not present in the condition ECT `environment-map` then the presence of, and value of, the corresponding ACS entry field does not affect whether the `environment-map`s are subsets.
+
+Before performing the binary comparison, a Verifier SHOULD convert `environment-map` fields into a form which meets CBOR Core Deterministic Encoding Requirements {{-cbor}}.
+
+### Mkey comparison {#sec-compare-mkey}
+
+The verifier SHALL iterate over all entries in the candidate entries array and compare the condition ECT `measurement-map/mkey` value to the candidate entry `measurement-map/mkey` value.
+If the two `mkey` values are not binary equal then the candidate entry is removed from the candidate entries array.
+
+Before performing the binary comparison, a Verifier SHOULD convert `mkey` fields into a form which meets CBOR Core Deterministic Encoding Requirements {{-cbor}}.
+
+If the `mkey` field is not present in neither the condition ECT nor the candidate entry then the `mkey` are equal and the candidate entry is not removed.
+If the `mkey` field is not present in only one side of the comparison then the `mkey` are not equal and the candiate entry is removed.
+
+### Authority comparison
+
+The verifier SHALL iterate over all entries in the candidate entries array and compare the condition ECT expected authority (`a`) value to the candidate entry authority value.
```

Comment:
Same comment as 2087

Reply:
Right, there is not a single condition authority, but an ANDed list. Each authority must have a matching key in the candidate ECT's authorities.

Reply:
In the internal representation, each ECT has its own authority